backend/docs/openapi.yaml

```
@@ -0,0 +1,56 @@
+openapi: "3.0.3"
+info:
+  title: JiroNow API
+  version: "1.0"
+
+paths:
+  /status:
+    get:
+      description: "Return the status of a store based on thr specified account's tweets"
+      parameters:
+        - name: accountId
+          in: query
+          required: true
+          schema:
+            type: string
+          description: "Store's Twitter account ID"
+      responses:
+        "200":
+          $ref: "#/components/responses/GetStoreStatusResponse"
+        "400":
+          $ref: "#/components/responses/InvalidAccountResponse"
+        "500":
+          $ref: "#/components/responses/UnprocessableStoreStatusResponse"
```

既知のエラーは500にしないでおきましょう

```suggestion
        "520":
          $ref: "#/components/responses/UnprocessableStoreStatusResponse"
         default:
             InternalServerError
```

500は本当に予期しないエラーに対してのみ使われるべきです

---

ちなみに、500系とかはわりと独自でステータスコード作っても良いです

MDNに載ってない範囲でエラーは切り出すとよいです

エラーメッセージはあくまで表示、デバッグのための値なので、ロジックには絡められません。フロントで特殊な処理(再送、特定のメッセージを表示)をしたい場合の分岐にはステータスコードしか使うべきではないため、500にいろいろなエラーを内包するのはよろしくないです

https://developer.mozilla.org/ja/docs/Web/HTTP/Status#%E3%82%B5%E3%83%BC%E3%83%90%E3%83%BC%E3%82%A8%E3%83%A9%E3%83%BC%E3%83%AC%E3%82%B9%E3%83%9D%E3%83%B3%E3%82%B9

---

これ知れてよかった、ありがとう